From d74d36cd2fbb523ce8521fdd6f9f904af1e42144 Mon Sep 17 00:00:00 2001
From: Rakesh Kudurumalla <rkudurumalla@marvell.com>
Date: Mon, 24 Apr 2023 10:21:55 +0530
Subject: [PATCH 382/955] app/test: resolve mbuf_test application failure

when RTE_ENABLE_ASSERT is defined test_mbuf application is
failing because we are trying to attach extbuf to a cloned
buffer to which external mbuf is already attached.To make
test_mbuf pass CI we have updated ol_flags. This patch fixes
the same.

Signed-off-by: Rakesh Kudurumalla <rkudurumalla@marvell.com>
Change-Id: Ia5a0f8a81036e12a1cfec82e61bfd6f1d0c6f89e
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/102082
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
Tested-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
---
 app/test/test_mbuf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/app/test/test_mbuf.c b/app/test/test_mbuf.c
index 53fe898a384cf..cfcb7c7dff2b0 100644
--- a/app/test/test_mbuf.c
+++ b/app/test/test_mbuf.c
@@ -2375,6 +2375,7 @@ test_pktmbuf_ext_shinfo_init_helper(struct rte_mempool *pktmbuf_pool)
 		GOTO_FAIL("%s: Bad packet length\n", __func__);
 
 	/* attach the same external buffer to the cloned mbuf */
+	clone->ol_flags = 0;
 	rte_pktmbuf_attach_extbuf(clone, ext_buf_addr, buf_iova, buf_len,
 			ret_shinfo);
 	if (clone->ol_flags != RTE_MBUF_F_EXTERNAL)
-- 
2.25.1

